### ISSUE TITLE
Error tracing miniF2F

### ISSUE THREAD

**Issue by Some-random** [lean-dojo/LeanDojo#2]
I'm trying to trace miniF2F, by running this code:
```
from lean_dojo import LeanGitRepo, trace
import os

repo = LeanGitRepo("https://github.com/openai/miniF2F", "4e433ff5cadff23f9911a2bb5bbab2d351ce5554")
trace(repo, dst_dir="miniF2F")
```

But I'm getting this error. I tried multiple times and the program always exits at 1159/4182:
```
2023-07-02 01:58:24.370 | INFO     | __main__:main:143 - Tracing miniF2F
 28%|██▊       | 1159/4182 [35:01<11:01:55, 13.14s/it]2023-07-02 02:34:04,667	INFO worker.py:1627 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265
  0%|          | 0/1159 [00:00<?, ?it/s](raylet) [2023-07-02 02:34:34,598 E 1108020 1108083] (raylet) agent_manager.cc:135: The raylet exited immediately because the Ray agent failed. The raylet fate shares with the agent. This can happen because the Ray agent was unexpectedly killed or failed. Agent can fail when
(raylet) - The version of `grpcio` doesn't follow Ray's requirement. Agent can segfault with the incorrect `grpcio` version. Check the grpcio version `pip freeze | grep grpcio`.
(raylet) - The agent failed to start because of unexpected error or port conflict. Read the log `cat /tmp/ray/session_latest/logs/dashboard_agent.log`. You can find the log file structure here https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure.
(raylet) - The agent is killed by the OS (e.g., out of memory).
```

I'm running on slurm, with native backend (I think docker is not allowed on my environment) and the machine has 450GB+ memory so memory shouldn't be an issue. I searched online for related error messages and tried to set some environment variables like cpu_count, OPENBLAS_NUM_THREADS, GOTO_NUM_THREADS and OMP_NUM_THREADS but got no luck...

**Comment by yangky11** [comment]
Could you please try 
```
import ray
ray.init()
```
to see if anything is wrong with Ray?

**Comment by Some-random** [comment]
As it turns out, ray.init() does produce the same error. But it works when I limit the number of CPUs for ray.init().

```
dojo >>> python
Python 3.10.11 (main, May 16 2023, 00:28:57) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ray
>>> ray.init(num_cpus=20)
2023-07-02 04:59:50,418	INFO worker.py:1627 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8266
RayContext(dashboard_url='127.0.0.1:8266', python_version='3.10.11', ray_version='2.5.1', ray_commit='a03efd9931128d387649dd48b0e4864b43d3bfb4', address_info={'node_ip_address': '172.31.130.195', 'raylet_ip_address': '172.31.130.195', 'redis_address': None, 'object_store_address': '/tmp/ray/session_2023-07-02_04-59-47_921783_1221083/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2023-07-02_04-59-47_921783_1221083/sockets/raylet', 'webui_url': '127.0.0.1:8266', 'session_dir': '/tmp/ray/session_2023-07-02_04-59-47_921783_1221083', 'metrics_export_port': 49046, 'gcs_address': '172.31.130.195:47238', 'address': '172.31.130.195:47238', 'dashboard_agent_listen_port': 52365, 'node_id': 'e25644ea3db265681439bdd8e58a608d93ba82e4eda6436d5d25921e'})
>>> ray.shutdown()
>>> ray.init()
2023-07-02 05:01:25,463	INFO worker.py:1627 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8266
[2023-07-02 05:01:55,602 E 1221083 1221083] core_worker.cc:193: Failed to register worker 01000000ffffffffffffffffffffffffffffffffffffffffffffffff to Raylet. IOError: [RayletClient] Unable to register worker with raylet. No such file or directory
```

I'll try to manually set the number of CPUs to 20 [here](https://github.com/lean-dojo/LeanDojo/blob/main/src/lean_dojo/utils.py#L73) to see whether it resolves my problem

**Comment by yangky11** [comment]
I haven't encountered this error before, though I suspect it's related to running Ray on your specific platform. There isn't much I can do about it on LeanDojo's side. Here are a few things you may want to try:
* Remove Ray's temporary directory (`/tmp/ray` by default) and re-try.
* Ask Ray people for help.
* Disable Ray in LeanDojo by setting `MAX_NUM_PROCs=1`. However, do expect it to be slower.
* If you want, I can share you the traced repo `LeanGitRepo("https://github.com/openai/miniF2F", "4e433ff5cadff23f9911a2bb5bbab2d351ce5554")`. Traced repos on different machines are interchangeable. However, since Ray is used by LeanDojo in many places, you're likely to run into other Ray-related errors if the problem is not resolved.

**Comment by Some-random** [comment]
Hi @yangky11, thanks for the info and I've figured out the problem!

So I've changed some of the config in my local LeanDojo code but it didn't have any effect because I installed LeanDojo with pip. After I reinstalled LeanDojo with `pip install .`, the change took effect and I was able to trace miniF2F.

I do have some other questions about miniF2F version and other traced Lean repo. I'll open another issue for it.
